tests/D2L.CodeStyle.Analyzers.Test/Async/Generator/SourceFilePathHintNameGeneratorTests.cs

```diff
@@ -0,0 +1,94 @@
+﻿using System.Collections.Immutable;
+using NUnit.Framework;
+
+namespace D2L.CodeStyle.Analyzers.Async.Generator;
+
+[TestFixture]
+internal class SourceFilePathHintNameGeneratorTests {

```

Comment:
Consider adding some cases with forward slashes as well, to make sure we're good for non-Windows systems.

Also, totally silly scenario, but consider adding a case for files that are on different drives as well. Could happen if, for example, source files were pulled from a NuGet package cached on C: while building from a checkout on D:.